src/routes/transactions/mappers/common/custom-transaction.mapper.ts

```diff
@@ -0,0 +1,84 @@
+import { Injectable } from '@nestjs/common';
+import { ModuleTransaction } from '../../../../domain/safe/entities/module-transaction.entity';
+import {
+  MultisigTransaction,
+  isMultisigTransaction,
+} from '../../../../domain/safe/entities/multisig-transaction.entity';
+import { AddressInfoHelper } from '../../../common/address-info/address-info.helper';
+import { NULL_ADDRESS } from '../../../common/constants';
+import { CustomTransactionInfo } from '../../entities/custom-transaction.entity';
+
+@Injectable()
+export class CustomTransactionMapper {
+  private static readonly MULTI_SEND = 'multiSend';
+  private static readonly TRANSACTIONS = 'transactions';
+  constructor(private readonly addressInfoHelper: AddressInfoHelper) {}
+
+  async mapCustomTransaction(
+    transaction: MultisigTransaction | ModuleTransaction,
+    value: number,
+    dataSize: number,
+    chainId: string,
+  ): Promise<CustomTransactionInfo> {
+    const toAddressInfo = await this.addressInfoHelper.getOrDefault(
+      chainId,
+      transaction.to,
+    );
+
+    return new CustomTransactionInfo(
+      toAddressInfo,
+      dataSize.toString(),
+      value.toString(),
+      transaction?.dataDecoded?.method ?? null,
+      this.getActionCount(transaction),
+      this.isCancellation(transaction, dataSize),
+    );
+  }
+
+  private getActionCount(
+    transaction: MultisigTransaction | ModuleTransaction,
+  ): number | null {
+    const { dataDecoded } = transaction;
+    if (dataDecoded?.method === CustomTransactionMapper.MULTI_SEND) {
+      const parameter = dataDecoded.parameters?.find(
+        (parameter) => parameter.name === CustomTransactionMapper.TRANSACTIONS,
+      );
+      return parameter?.valueDecoded?.length ?? null;
+    }
+
+    return null;
+  }
+
+  private isCancellation(
+    transaction: MultisigTransaction | ModuleTransaction,
+    dataSize: number,
+  ): boolean {
+    if (isMultisigTransaction(transaction)) {
```

Comment:
Here I think is clearer to check if the transaction has been executed instead of checking the type of the transaction. I mean, in the future we could have another type of transaction that should be included in this logic like `... || isXTypeTransaction(transaction) || isYTypeTransaction(transaction) || ...`, etc.

Instead of that, I think it's clearer to check if the `isExecuted` property is in the transaction by using a type guard, and that way the `isMultisigTransaction` function could be removed. What do you think about using something like this instead:

```javascript
if ('isExecuted' in transaction) {
...
} else {
return false;
}
```

Reply:
Makes sense, done here https://github.com/5afe/safe-client-gateway-nest/pull/205/commits/98a846361bde9966b278db3c6dff9b5a71a78278